docs/src/tutorials/software-management.md

```diff
@@ -0,0 +1,120 @@
+
+# Introduction
+
+This document describes how to install and use the **Software management** feature in thin-edge.io.
+
+> Note: As of now the software management only supports **debian** based distros(Ex: Ubuntu, Debian),
+ which use the **apt**.
+
+
+
+## Download and Install software management packages on the device
+
+As a prequisite, install [tedge and tedge_mapper](../howto-guides/002_installation.md) if not installed already. 
+
+The thin-edge software management packages are in repository on GitHub: [thin-edge.io](https://github.com/thin-edge/thin-edge.io/releases).
+
+To download the package from github repository use the following command (use desired version):
+
+```shell
+curl -LJO https://github.com/thin-edge/thin-edge.io/releases/download/<package>_<version>_<arch>.deb
+```
+
+where:
+> `version` -> thin-edge.io software management components version in x.x.x format
+>
+> `arch` -> architecture type (amd64, armhf)
+
+Download `tedge_apt_plugin` and `tedge_agent`
+
+```shell
+curl -LJO https://github.com/thin-edge/thin-edge.io/releases/download/0.3.0/tedge_apt_plugin_0.3.0_amd64.deb
+curl -LJO https://github.com/thin-edge/thin-edge.io/releases/download/0.3.0/tedge_agent_0.3.0_amd64.deb
+```
+
+Now the packages are downloaded, proceed to installation.
+> Note: Some OSes may require you to use `sudo` to install packages and therefore all following commands may need `sudo`.
+
+To install `tedge_apt_plugin` and `tedge_agent` on thin-edge device do:
+
+```shell
+dpkg -i tedge_apt_plugin_<version>_<arch>.deb
+dpkg -i tedge_agent<version>_<arch>.deb
+```
+
+
+
+## Start and enable the software management services
+
+The `tedge connect` will automatically start and enable the software management services.
+Find more about [how to connect thin-edge device to cloud](../howto-guides/004_connect.md)
+
+Once the thin-edge device is successfully connected to Cumulocity cloud, the **Software** option will be enabled and
+the list of softwares that are installed on the device will be visible as shown in the figure below.
+
+![Add new software](./images/start-software-management.png)
+
+
+> Note: Disconnecting thin-edge device from cloud will stop and disable the software management services.
+
+
+
+## Managing the device software on Cumulocity
+
+More info about managing the device software can be found [here](https://cumulocity.com/guides/users-guide/device-management/#managing-device-software)
+
+### Adding new software into software repository
+
+1. In the Software repository page, click Add software at the right of the top menu bar.
```

Comment:
It would be better if we specify how to get to the software repository page from here. You could highlight that option on the left hand side pane in the UI in the image that you've linked below.